cartridges/int_algolia/cartridge/scripts/algolia/steps/sendChunkOrientedDeltaProductUpdates.js

```diff
@@ -0,0 +1,336 @@
+const updateLogType = 'LastProductDeltaSyncLog';
+const resourceType = 'productdelta';
+
+var l0_deltaExportDir, l1_processingDir, l1_completedDir;
+var changedProducts = [], changedProductsIterator;
+var deltaExportZips, siteLocales, nonLocalizedAttributes = [], fieldsToSend;
+var baseIndexingOperation; // 'addObject' or 'partialUpdateObject', depending on the step parameter 'fullRecordUpdate'
+const deleteIndexingOperation = 'deleteObject';
+
+/*
+ * Rough algorithm of chunk-oriented script module execution:
+ *
+ *  counter = 0
+ *  beforeStep() - executed before starting the job step, initialization
+ *    repeat until counter reaches getTotalCount() {
+ *         create new thread {
+ *             beforeChunk() - executed before each chunk
+ *             read() - repeat until chunkSize, each run reads one product
+ *             process() - repeat until chunkSize, each run processes one product
+ *             write() - executed one time, sends all products from the chunk at once
+ *             counter += chunkSize
+ *             afterChunk() - executed after each chunk
+ *         }
+ *     }
+ *     afterStep() - executed after the job step finishes
+ */
+
+/**
+ * before-step-function (steptypes.json)
+ * Any returns from this function result in skipping to the afterStep() function (omitting read-process-writealtogether)
+ * with the "success" parameter passed to it set to false.
+ * @param {dw.util.HashMap} parameters job step parameters
+ * @param {dw.job.JobStepExecution} stepExecution contains information about the job step
+*/
+exports.beforeStep = function(parameters, stepExecution) {
+    algoliaData = require('*/cartridge/scripts/algolia/lib/algoliaData');
+    AlgoliaLocalizedProduct = require('*/cartridge/scripts/algolia/model/algoliaLocalizedProduct');
+    algoliaProductConfig = require('*/cartridge/scripts/algolia/lib/algoliaProductConfig');
+    jobHelper = require('*/cartridge/scripts/algolia/helper/jobHelper');
+    fileHelper = require('*/cartridge/scripts/algolia/helper/fileHelper');
+    algoliaIndexingAPI = require('*/cartridge/scripts/algoliaIndexingAPI');
+    logger = require('dw/system/Logger').getLogger('algolia', 'Algolia');
+    productFilter = require('*/cartridge/scripts/algolia/filters/productFilter');
+    sendHelper = require('*/cartridge/scripts/algolia/helper/sendHelper');
+
+    // checking mandatory parameters
+    if (empty(parameters.consumer) || empty(parameters.deltaExportJobName)) {
+        let errorMessage = 'Mandatory job step parameters missing!';
+        jobHelper.logError(errorMessage);
+        return;
+    }
+
+    // parameters
+    paramConsumer = parameters.consumer.trim();
+    paramDeltaExportJobName = parameters.deltaExportJobName.trim();
+    paramFieldListOverride = algoliaData.csvStringToArray(parameters.fieldListOverride); // fieldListOverride - pass it along to sendChunk()
+    baseIndexingOperation = !!parameters.fullRecordUpdate ? 'addObject' : 'partialUpdateObject';
```

Comment:
I wonder if this parameter is useful for a delta export, the job could only send partial updates no? I'll add it to the discussion.